libraries/lwIP_enc28j60/src/utility/enc28j60.cpp

```diff
@@ -742,3 +766,106 @@ ENC28J60::readFrameData(uint8_t *buffer, uint16_t framesize)
 
     return _len;
 }
+
+/*---------------------------------------------------------------------------*/
+uint16_t
+ENC28J60::phyRead(uint8_t address)
+{
+    unsigned int            timeout = 15;
+
+    setregbank(REG_BANK_2);
+    
+    // set the PHY register address
+    writereg(MIREGADR, address);
+    setregbitfield(MICMD, 0x01);                // MII Read Enable [MIIRD]
+    
+    setregbank(REG_BANK_3);
+	
+    // wait until the PHY read completes
+    while(readreg(MISTAT) & 0x01){
+        delay(10);
+        wdt_reset();
+        if(--timeout ==0)       return 0;
+    }
+    
+    setregbank(REG_BANK_2);
+    
+    clearregbitfield(MICMD, 0x01);              // MII Read Disable [MIIRD]
+    
+    // read the PHY data
+    return (readreg(MIRDL) | readreg(MIRDH) << 8);
+}
+
+void
+ENC28J60::phyWrite(uint8_t address, uint16_t data)
+{
+    unsigned int            timeout = 15;
+    
+    setregbank(REG_BANK_2);
+    
+    // set the PHY register address
+    writereg(MIREGADR, address);
+    
+    // write the PHY data
+    writereg(MIWRL, data & 0xff);
+    writereg(MIWRH, data >> 8);
+    
+    setregbank(REG_BANK_3);
+    
+    // wait until the PHY write completes
+    while(readreg(MISTAT) & 0x01){
+        delay(10);
+        wdt_reset();
+        if(--timeout ==0)       return;
+    }
+}
+
+//--------------------------------------------------------
+void
+ENC28J60::init(void)
+{
+    pinMode(_cs, OUTPUT);
+    digitalWrite(_cs, HIGH);
+    SPI.begin();
+}
+
+void
+ENC28J60::LEDsConfig(LEDScfg_CONFIG LEDAcfg, LEDScfg_CONFIG LEDBcfg)
+{
+    uint16_t                temp  = phyRead(PHLCON);
+    
+    if(LEDAcfg)             temp &= LEDA_MASK;
+    if(LEDBcfg)             temp &= LEDB_MASK;
+	
+    phyWrite(PHLCON, (temp |(uint16_t)(LEDAcfg <<4) |LEDBcfg |0x3006));
+}
+
+//--------------------------------------------------------
+bool
+ENC28J60::resetEther(void)
+{
+    return reset();
+}
+
+uint8_t
+ENC28J60::hardwareStatus(void)
```

Comment:
Is it really `hardwareStatus()` or hardwareRevision/hardwareVersion() ?
We already have [`::readrev()`](https://github.com/esp8266/Arduino/blob/bf2882d8e003c278017639fa964248fedb942454/libraries/lwIP_enc28j60/src/utility/enc28j60.cpp#L308L322) which seems to be identical.

Reply:
- This function is similar to readrev function (it is private)
- If this function returns 0xFF, that means "Hardware Error" (incorrect hardware revision)

Reply:
This function can be changed this way
```
uint8_t
ENC28J60::hardwareStatus(void)
{
	return readrev();		/* 0xFF = Hardware Error */
}
```

Reply:
Or it could be, as for `reset()`, moved to the public section without changing its name ?